kpops/utils/dict_ops.py

```diff
@@ -77,7 +77,8 @@ def generate_substitution(
     input: dict,
     prefix: str | None = None,
     existing_substitution: dict | None = None,
-) -> dict:
+    separator: str | None = None,
+) -> dict[Any, Any]:
```

Comment:
isn't it essentially `dict[_K, _V]` same as the input dictionary? if it's not needed I would remove the `[Any, Any]`

Reply:
It is. Since the mapping flattening function requires `str` keys, I further narrowed it down to `[str, _V]`